backend/src/main/java/com/dccsacademy/repositories/DepartmentRepository.java

```diff
@@ -0,0 +1,13 @@
+package com.dccsacademy.repositories;
+
+import com.dccsacademy.entities.DepartmentEntity;
+import io.quarkus.hibernate.orm.panache.PanacheRepository;
+import jakarta.enterprise.context.ApplicationScoped;
+
+@ApplicationScoped
+public class DepartmentRepository implements PanacheRepository<DepartmentEntity> {
+
+    public DepartmentEntity findByName(String name) {
+        return find("name", name).firstResult();
```

Comment:
I know this works, but you should use a chance and try to implement it over Criteria API query.